packages/core/src/servient.ts

```diff
@@ -210,18 +213,30 @@ export default class Servient {
 
     // will return WoT object
     public async start(): Promise<typeof WoT> {
+        if (this.started) {
+            throw Error("Servient started already");
+        }
         const serverStatus: Array<Promise<void>> = [];
         this.servers.forEach((server) => serverStatus.push(server.start(this)));
         this.clientFactories.forEach((clientFactory) => clientFactory.init());
 
         await Promise.all(serverStatus);
+        this.started = true;
         return new WoTImpl(this);
     }
 
     public async shutdown(): Promise<void> {
-        this.clientFactories.forEach((clientFactory) => clientFactory.destroy());
+        if (this.started) {
+            if (this.shutdowned) {
+                throw Error("Servient shutdowned already");
+            }
```

Comment:
I think by reversing the logic here, we could get rid of one level of nesting and make the code a bit more readable.

```suggestion
        if (!this.started) {
            debug(`Servient cannot be shut down, wasn't even started`);
            return;
        }
        
        if (this.shutdowned) {
            throw Error("Servient shutdowned already");
        }
```

Reply:
Sometimes I think people have different _taste_ of code. I am mostly fine with nesting while you dislike it ;-)
Anyhow, in this case I am fine either way. Fixed with https://github.com/eclipse-thingweb/node-wot/pull/1195/commits/b2b4388aeb71135e178769faa7a50008ad08f664